packages/loader/container-loader/src/containerContext.ts

```diff
@@ -49,66 +48,84 @@ type ContextLifecycleEvents = "runtimeInstantiated" | "disposed";
 
 export class ContainerContext implements IContainerContext {
 	public static async createOrLoad(
-		container: Container,
+		options: ILoaderOptions,
 		scope: FluidObject,
 		runtimeFactory: IRuntimeFactory,
 		baseSnapshot: ISnapshotTree | undefined,
+		version: IVersion | undefined,
 		deltaManager: IDeltaManager<ISequencedDocumentMessage, IDocumentMessage>,
+		storage: IDocumentStorageService,
 		quorum: IQuorum,
+		audience: IAudience,
 		loader: ILoader,
 		submitFn: (type: MessageType, contents: any, batch: boolean, appData: any) => number,
 		submitSummaryFn: (summaryOp: ISummaryContent, referenceSequenceNumber?: number) => number,
 		submitBatchFn: (batch: IBatchMessage[], referenceSequenceNumber?: number) => number,
 		submitSignalFn: (contents: any) => void,
 		disposeFn: (error?: ICriticalContainerError) => void,
 		closeFn: (error?: ICriticalContainerError) => void,
-		version: string,
 		updateDirtyContainerState: (dirty: boolean) => void,
+		getAbsoluteUrl: (relativeUrl: string) => Promise<string | undefined>,
+		getContainerDiagnosticId: () => string | undefined,
+		getClientId: () => string | undefined,
+		getServiceConfiguration: () => IClientConfiguration | undefined,
+		getAttachState: () => AttachState,
+		clientDetails: IClientDetails,
 		existing: boolean,
+		connected: boolean,
+		taggedLogger: ITelemetryLoggerExt,
 		pendingLocalState?: unknown,
 	): Promise<ContainerContext> {
 		const context = new ContainerContext(
-			container,
+			options,
 			scope,
 			runtimeFactory,
 			baseSnapshot,
+			version,
 			deltaManager,
+			storage,
 			quorum,
+			audience,
 			loader,
 			submitFn,
 			submitSummaryFn,
 			submitBatchFn,
 			submitSignalFn,
 			disposeFn,
 			closeFn,
-			version,
 			updateDirtyContainerState,
+			getAbsoluteUrl,
+			getContainerDiagnosticId,
+			getClientId,
+			getServiceConfiguration,
+			getAttachState,
+			clientDetails,
 			existing,
+			connected,
+			taggedLogger,
 			pendingLocalState,
 		);
 		await context.instantiateRuntime(existing);
 		return context;
 	}
 
-	public readonly taggedLogger: ITelemetryLoggerExt;
 	public readonly supportedFeatures: ReadonlyMap<string, unknown>;
 
 	public get clientId(): string | undefined {
-		return this.container.clientId;
+		return this._getClientId();
 	}
 
 	/**
 	 * DISCLAIMER: this id is only for telemetry purposes. Not suitable for any other usages.
 	 */
 	public get id(): string {
-		return this.container.resolvedUrl?.id ?? "";
+		return this._getContainerDiagnosticId() ?? "";
```

Comment:
we should just deprecate this. we already add it to the logger that gets passed to the runtime, and no one but the runtime has access to this interface, so i don't really see a use for it. host may need it, but they can and should only get it off container.

Reply:
I'd agree.  This PR doesn't touch the IContainerContext interface - would probably do a separate PR focused on IContainerContext changes, since we'll need to think hard about staging given the layer cross.